File: C++/Binary Search.cpp

		#include<bits/stdc++.h>
using namespace std;

int binarysearch(int arr[],int l,int h,int key)
{

	while(l<=h)
	{
		int mid=l+(h-l)/2;
		if(arr[mid]<key)
		l=mid+1;
		else if(arr[mid]==key)
		{
		 return mid;
		}
		else h=mid-1;

	}
	return -1;
}


int main()
{
	int n;
	cin>>n;
	int arr[n];
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	cout<<"enter element to search"<<endl;
	int key;
	cin>>key;
	cout<<binarysearch(arr,0,n-1,5);
	
	
/*	int l=0;
	int h=n-1;
	int mid=l+(h-l)/2;
	while(l<=h)
	{
		if(arr[mid]<key)
		{
			l=mid+1;
		}
		else if(arr[mid]==key)
		{
		cout<<"element found at position "<<mid;
		break;
		}
		else 
		{
			h=mid-1;
		}
		mid=l+(h-l)/2;
		
		if(l>h)
		cout<<"element not found";*/
	}

File: Java/binarySearch.java
public class Main
{
  public static void main (String[]args)
  {
    int[] arr={0,2,5,88,100};
    int target=88;
    int ans=binarySearch(arr,target);
   System.out.println(ans);
  }
  static int binarySearch (int[]arr, int target)
  {
    int start = 0;
    int end = arr.length - 1;
    while (start <= end)
      {
	//  int mid=(start+end)/2; //[probelm coz of int]
	int mid = start + (end - start) / 2;
	if (target < arr[mid])
	  {
	    end = mid - 1;
	  }
	else if (target > arr[mid])
	  {
	    start = mid + 1;
	  }
	else
	  {
	    return mid;
	  }
      }
      return -1;
  }
}

//output:
//3
